components/eamxx/scripts/cime-nml-tests

```diff
@@ -0,0 +1,276 @@
+        for item in self._dirs_to_cleanup:
+            shutil.rmtree(item)
+
+    ###########################################################################
+    def test_doctests(self):
+    ###########################################################################
+        """
+        Run doctests for all eamxx/cime_config python files and nml-related files in scripts
+        """
+        run_cmd_assert_result(self, "python3 -m doctest *.py", from_dir=EAMXX_CIME_DIR)
+        run_cmd_assert_result(self, "python3 -m doctest atm_manip.py", from_dir=EAMXX_SCRIPTS_DIR)
+
+    ###########################################################################
+    def test_pylint(self):
+    ###########################################################################
+        """
+        Run pylint on all eamxx/cime_config python files and nml-related files in scripts
+        """
+        ensure_pylint()
+        run_cmd_assert_result(self, "python3 -m pylint --disable C,R,E0401,W1514 *.py", from_dir=EAMXX_CIME_DIR)
+        run_cmd_assert_result(self, "python3 -m pylint --disable C,R,E0401,W1514 atm_manip.py", from_dir=EAMXX_SCRIPTS_DIR)
+
+    ###########################################################################
+    def test_xmlchange_propagates_to_atmconfig(self):
+    ###########################################################################
+        """
+        Test that xmlchanges impact atm config files
+        """
+        case = self._create_test("ERS_Ln22.ne30_ne30.F2010-SCREAMv1 --no-build".split())
+
+        # atm config should match case test opts
+        case_rest_n = run_cmd_assert_result(self, "./xmlquery REST_N --value", from_dir=case)
+        atm_freq    = run_cmd_assert_result(self, "./atmquery Frequency --value", from_dir=case)
+        self.assertEqual(case_rest_n, atm_freq)
+
+        # Change XML and check that atmquery reflects this change
+        new_rest_n = "6"
+        self.assertNotEqual(new_rest_n, case_rest_n)
+        run_cmd_assert_result(self, f"./xmlchange REST_N={new_rest_n}", from_dir=case)
+        run_cmd_assert_result(self, "./case.setup", from_dir=case)
+        new_case_rest_n = run_cmd_assert_result(self, "./xmlquery REST_N --value", from_dir=case)
+        new_atm_freq    = run_cmd_assert_result(self, "./atmquery Frequency --value", from_dir=case)
+        self.assertEqual(new_case_rest_n, new_rest_n)
+        self.assertEqual(new_atm_freq, new_rest_n)
+
+    ###########################################################################
+    def test_atmchanges_are_preserved(self):
+    ###########################################################################
+        """
+        Test that atmchanges are not lost when eamxx setup is called
+        """
+        case = self._create_test("SMS.ne30_ne30.F2010-SCREAMv1 --no-build".split())
+
+        bm = run_cmd_assert_result(self, "./atmquery bm --value", from_dir=case)
```

Comment:
I would pick something that is *always* going to be there in the future. `bm` is there as part of the _current_ way we init that field (namely, equal to a constant). If that changes, the parameter `bm` may not be found in the yaml file.

I would pick something like `atm_log_level`, which we're quite sure will always be there.